contracts/sol/MACI.sol

```diff
@@ -178,15 +177,24 @@ contract MACI is Ownable, DomainObjs {
         // which has no meaning attached to it
         voteOptionsMaxLeafIndex = _maxValues.maxVoteOptions;
 
+        // Create the message tree
+        messageTree = new IncrementalMerkleTree(_treeDepths.messageTreeDepth, ZERO_VALUE);
+
         // Calculate and store the empty vote option tree root. This value must
         // be set before we call hashedBlankStateLeaf() later
         emptyVoteOptionTreeRoot = calcEmptyVoteOptionTreeRoot(_treeDepths.voteOptionTreeDepth);
-        currentResultsCommitment = hashZeros(2 ** uint256(_treeDepths.voteOptionTreeDepth) + 1);
 
+        // Calculate and store a commitment to 2 ** voteOptionTreeDepth zeros,
+        // and the salt of 0.
+        currentResultsCommitment = hashLeftRight(
+            computeEmptyRoot(_treeDepths.voteOptionTreeDepth, 0),
```

Comment:
This should be `emptyVoteOptionTreeRoot`?

```suggestion
            emptyVoteOptionTreeRoot,
```

Reply:
Oops!  Computing the root twice wastes gas. Thank you for spotting this. I'll push a commit to fix it.